crates/task-impls/src/transactions.rs

```diff
@@ -233,15 +233,13 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                 // changes.
                 // TODO <https://github.com/EspressoSystems/HotShot/issues/1693>
                 let vid = VidScheme::new(NUM_CHUNKS, NUM_STORAGE_NODES, &srs).unwrap();
-                let vid_disperse = vid
-                    .disperse(encoded_txns.into_iter().collect::<Vec<u8>>())
-                    .unwrap();
+                let vid_disperse = vid.disperse(encoded_transactions.clone()).unwrap();
 
                 // TODO never clone a block
                 // https://github.com/EspressoSystems/HotShot/issues/1858
                 self.event_stream
                     .publish(HotShotEvent::BlockReady(
-                        payload.clone(),
+                        encoded_transactions.into_iter().collect::<Vec<u8>>(),
```

Comment:
It looks like above on line 224 we already collected `encoded_transactions` into a `Vec<u8>`. Unless I'm missing something in between this `into_iter().collect()` looks like a noop.

Reply:
I think you are right! I thought lint would catch things like this. 🤔

Reply:
Yeah me too...weird